### PR TITLE
AEA-1063 win publish tar remove fix

### LINKED ISSUE #1940: PermissionError: [WinError 32] during publishing of agent for AW2 Testnet
**Describe the bug**
Hi everyone;
I get the following error when I try to publish my skills for agentland phase II for the testnet:
PermissionError: [WinError 32] Der Prozess kann nicht auf die Datei zugreifen, da sie von einem anderen Prozess verwendet wird: 'C:\\Windows\\System32\\registration_aea_aw1\\aw2\\aw2.tar.gz'

From what I read and understood, there seems to be a problem with Python that can occur under Windows as in that there is a command missing to "remove" the using of a created tar.gz file (or an image, or any other kind of file). Therefore, the console is preventing itself from further using that file. I ccannot, therefore, at the moment, publish my agents and havent been able to do so.

I am running the following system:
Windows 10 Pro laptop, Python 3.8; AEA v0.7.3

**To Reproduce**
1. I have generated the 2 skills simple_seller and simple_data_request. My transactions have been successfully run, and everything should be ready to publish according to the steps for Testnet Phase 2. 
Then:
I open a console with admin rights.
1. I go to aw2 Folder in console (name of my agent)
2. I type aea publish --remote after having pushed both skills (which was possible)
3. the error as below pops up. However, 3 tar.gz files are visible and available in the aw2 folder in Windows Explorer.

**Expected behavior**
The expected behavior was to see the publishing of the agent as described in the instructions. 

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10pro
 - Browser not applicable in this case. It is Windows console
 - Version AEA 0.7.3 and Python 3.8

**Smartphone (please complete the following information):**

**Additional context**
I read in other forums (for instance here: https://stackoverflow.com/questions/27215462/permissionerror-winerror-32-the-process-cannot-access-the-file-because-it-is

I think this is a similar problem but i wasnt able to fix that code in the python environment. 

screenpythonerror

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

Error during package publish triggers exception on tar file remove on windows platform

## Fixes

#1940

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/registry/publish.py [hunk 1]
```diff
@@ -76,13 +76,14 @@ def publish_agent(ctx: Context):
         "skills": ctx.agent_config.skills,
     }
 
-    files = {"file": open(output_tar, "rb")}
-    if is_readme_present(readme_source_path):
-        files["readme"] = open(readme_source_path, "rb")
+    with open(output_tar, "rb") as fd:
+        files = {"file": fd}
+        if is_readme_present(readme_source_path):
+            files["readme"] = open(readme_source_path, "rb")
```
Comment (DavidMinarsch) [inline aea/cli/registry/publish.py:82]: better to do nested with? i.e. another context manager for readme_source_path?

**Review by DavidMinarsch — APPROVED** [review]
LGTM, one potential improvement

File: aea/cli/registry/publish.py (hunk 1 above)
Comment (solarw) [reply aea/cli/registry/publish.py:82]: missed this, thanks, but this readme  is not removed

**Review by marcofavorito — APPROVED** [review]
LGTM

**Pull request merged**